protocol/http.cpp

```diff
@@ -92,28 +59,40 @@ std::string Http::create_get_request(uri::Uri const &uri) {
     return ss.str();
 }
 
-Response Http::parse_response(std::string_view data) {
-    auto [header, body] = split(data, "\r\n\r\n");
-    auto status_line = parse_status_line(header);
-    if (!status_line) {
-        return {Error::InvalidResponse};
+std::optional<StatusLine> Http::parse_status_line(std::string_view status_line) {
+    auto sep1 = status_line.find(' ');
+    if (sep1 == std::string_view::npos) {
```

Comment:
Not needed, but you could also access `npos` from the `status_line` variable. That way you only need to change the argument type in one place if it ever changes. ...the argument type of this will never change, but it's a neat thing to be able to do. :P

Reply:
Fixed!